packages/nouns-contracts/contracts/governance/NounsDAOLogicV2.sol

```diff
@@ -0,0 +1,895 @@
+        emit NewPendingAdmin(oldPendingAdmin, pendingAdmin);
+    }
+
+    /**
+     * @notice Changes vetoer address
+     * @dev Vetoer function for updating vetoer address
+     */
+    function _setVetoer(address newVetoer) public {
+        require(msg.sender == vetoer, 'NounsDAO::_setVetoer: vetoer only');
+
+        emit NewVetoer(vetoer, newVetoer);
+
+        vetoer = newVetoer;
+    }
+
+    /**
+     * @notice Burns veto priviledges
+     * @dev Vetoer function destroying veto power forever
+     */
+    function _burnVetoPower() public {
+        // Check caller is pendingAdmin and pendingAdmin ≠ address(0)
+        require(msg.sender == vetoer, 'NounsDAO::_burnVetoPower: vetoer only');
+
+        _setVetoer(address(0));
+    }
+
+    /**
+     * @notice Current proposal threshold using Noun Total Supply
+     * Differs from `GovernerBravo` which uses fixed amount
+     */
+    function proposalThreshold() public view returns (uint256) {
+        return bps2Uint(proposalThresholdBPS, nouns.totalSupply());
+    }
+
+    function proposalCreationBlock(Proposal storage proposal) internal view returns (uint256) {
+        if (proposal.creationBlock == 0) {
+            return proposal.startBlock - votingDelay;
+        }
+        return proposal.creationBlock;
+    }
+
+    /**
+     * @notice Quorum votes required for a specific proposal to succeed
+     * Differs from `GovernerBravo` which uses fixed amount
+     */
+    function quorumVotes(uint256 proposalId) public view returns (uint256) {
+        Proposal storage proposal = _proposals[proposalId];
+        if (proposal.totalSupply == 0) {
+            return proposal.minQuorumVotes;
+        }
+
+        return
+            dynamicQuorumVotes(
+                proposal.againstVotes,
+                proposal.totalSupply,
+                getDynamicQuorumParamsAt(proposalCreationBlock(proposal))
+            );
+    }
+
+    /**
+     * @notice Calculates the required quorum of for-votes based on the amount of against-votes
+     *     The more against-votes there are for a proposal, the higher the required quorum is.
+     *     The quorum BPS is between `params.minQuorumVotesBPS` and params.maxQuorumVotesBPS.
+     *     `params.quorumVotesBPSOffset` determines at which amount of against-votes does the
+     *       additional quorum calculate "kick in".
+     *     The additional quorum is calculated as: coefs[0] * againstVotesBPS + coefs[1] * againstVotesBPS^2
+     *       where coefs is `params.quorumPolynomCoefs`
+     * @dev Note that the quorumPolynomCoefs are fixed point integers with 6 decimals
+     * @param againstVotes Number of against-votes in the proposal
+     * @param totalSupply The total supply of Nouns at the time of proposal creation
+     * @return quorumVotes The required quorum
+     */
+    function dynamicQuorumVotes(
+        uint256 againstVotes,
+        uint256 totalSupply,
+        DynamicQuorumParams memory params
```

Comment:
Add param comment for this one.

Reply:
fixed: https://github.com/nounsDAO/nouns-monorepo/pull/438/commits/807c3adb12ee7aa08c43d41dee6807a8dd53645f
If you think the description of the `DynamicQuorumParams` struct members should also go here, lmk